### ISSUE TITLE
Allow hiding info hints from validate output with a cli flag

### ISSUE THREAD

**Issue by bbhtt** [ximion/appstream#600]
Right now if you have a very big metainfo file, it outputs a huge list of info, warnings and errors.

The last two are to be fixed immediately while info hints are good to know, people might not be able to solve them immediately.

It'd be nice if the info hints starting with `I:` can be hidden with a cli flag.

**Comment by ximion** [comment]
If we hide them, people wouldn't see any new ones and would just plainly apply "that flag that makes the validator shut up". What would be examples of hints that couldn't be fixed, and would the ability to downgrade them to pedantic priority help?

**Comment by bbhtt** [comment]
The initial post was based off of this https://buildbot.flathub.org/#/builders/26/builds/12353/steps/11/logs/stdio but looking through the commit log yesterday I just realized that one was fixed. I found some more, but I don't have the links now.

Some maintainers don't want to use the new tag, because their upstream CI still uses, appstream-glib etc. 

And the actual error that needs to be fixed right now is `"W: org.gimp.GIMP:432: tag-empty id",` which gets hidden.

**Comment by ximion** [comment]
> Some maintainers don't want to use the new tag, because their upstream CI still uses, appstream-glib etc.

I'd see that as a bug that they need to fix ;-)

> And the actual error that needs to be fixed right now is `"W: org.gimp.GIMP:432: tag-empty id",` which gets hidden.

Woah, that's a pretty serious one!

On the `developer-name-tag-deprecated` issue, that one was supposed to be emitted only once, that issue was resolved in Git master. A new AppStream release is scheduled for tomorrow, so you will have far less noise very soon if you update your AS version. So that might already help you enough here.

What also helps: `appstreamcli` emits colored output by default. If you could show that, finding the issues that are actually important is trivial.

> Another example is screenshot URLs not being https, it's not something that can be solved immediately by an app maintainer (sometimes third party), if upstream doesn't provide https urls.

That's why it's only an info - it should be fixed eventually, but it will not fail validation. You could demote this issue to pedantic and hide it, but it would be good to keep it in the mind of people working on this, because:

> (Flathub recommends screenshots to be hosted by upstream)

Does that mean every SC is hitting *upstream* sites for screenshots, and also via a HTTP connection without encrption sometimes? That would have *huge* privacy implications!
If Flathub mirrors things though, it's fine, then it would only hit the Flathub CDN (I'll have a look at this when I'm home).

**Comment by bbhtt** [comment]
No, they are mirrored to Flathub's CDN which has https but that ends up in catalogue while validation is run on metainfo which has the original URLs.

**Comment by ximion** [comment]
That's fine then :-) - It's a nice thing if upstream URLs are also HTTPS (especially if usually both variants are provided and sometimes a http link is an accident), but it's not an error.
